src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1152,36 +1152,13 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
 
                     try
                     {
-                        var finalCt = CancellationToken.None;
+                        connector.ResetCancellation();
```

Comment:
nit: you can move this and the two locals outside the try, and then use them in the catch/finally below